mir_perception/mir_perception_utils/common/src/pointcloud_utils.cpp

```diff
@@ -51,41 +51,47 @@ unsigned int pointcloud::centerPointCloud(const PointCloud &cloud_in, PointCloud
 
 unsigned int pointcloud::padPointCloud(PointCloud::Ptr &cloud_in, int num_points)
 {
-  if (cloud_in->empty()) return (0);
+    if (cloud_in->empty()) 
+        return (0);
 
-  unsigned int point_count;
-  point_count = static_cast<unsigned int>(cloud_in->size());
+    unsigned int point_count;
+    point_count = static_cast<unsigned int>(cloud_in->size());
+
+    if (point_count > num_points) 
+    {
+        unsigned int point_diff = point_count - num_points;
 
-  if (point_count > num_points) {
-    unsigned int point_diff = point_count - num_points;
+        boost::random::mt19937 rd; 
+        boost::random::uniform_int_distribution<> distr(0,point_count);
 
-    std::random_device rd;   // get ran
```

Comment:
this doesn't work with c++14?

Reply:
No, it was not working. That's why I changed it to boost